pkg/distributor/distributor.go

```diff
@@ -857,6 +872,18 @@ func (d *Distributor) prePushValidationMiddleware(next push.Func) push.Func {
 		d.receivedRequests.WithLabelValues(userID).Add(1)
 		d.activeUsers.UpdateUserTimestamp(userID, now)
 
+		group := ""
+		if len(req.Timeseries) > 0 {
+			group = validation.FindGroupLabel(d.limits, userID, req.Timeseries[0].Labels)
+			if d.activeGroups.ActiveGroupLimitExceeded(userID, group) {
+				group = "other"
+			}
+		}
+
+		if group != "" {
+			d.activeGroups.UpdateGroupTimestamp(userID, group, now)
+		}
```

Comment:
Instead of doing two steps (checking if group limit is exceeded and then updating the timestamp) I would expect this to do everything in a single step, kind of like:

```suggestion
		group := ""
		if len(req.Timeseries) > 0 {
			group = validation.FindGroupLabel(d.limits, userID, req.Timeseries[0].Labels)
		}
		if group != "" {
			group = d.activeGroups.UpdateActiveGroup(userID, group, now)
		}
```